src/plugins/discover/public/application/components/sidebar/discover_field.tsx

```diff
@@ -311,6 +324,29 @@ export function DiscoverField({
             />
           </EuiFlexItem>
         )}
+        {canDeleteField && (
+          <EuiFlexItem grow={1} data-test-subj="discoverFieldListPanelDeleteItem">
```

Comment:
This flex group uses a medium gutter width which adds a lot of space between the buttons:
Discover:
<img width="314" alt="Screenshot 2021-04-13 at 11 44 33" src="https://user-images.githubusercontent.com/1508364/114532955-aafc2c00-9c4d-11eb-9d36-66c987837887.png">

Lens:
<img width="303" alt="Screenshot 2021-04-13 at 11 44 42" src="https://user-images.githubusercontent.com/1508364/114532964-acc5ef80-9c4d-11eb-9e44-219adb102a8a.png">

Reply:
Changed to `xs` gutter size